### ISSUE TITLE
the added layer must be an instance of Layer

### ISSUE THREAD

**Issue by rescuehero** [plaidml/plaidml#1431]
tldr;

> plaidml keras does throw: TypeError: The added layer must be an instance of class Layer.

I want to use plaidml with keras on Win10 using a AMD Radeon HD 7870 and Python 3.7

so I created a new anaconda venv, uninstalled vcred and reinstalled it, reinstalled my amd drivers and followed your steps:
```
pip install -U plaidml-keras
plaidml-setup
```
and
```
pip install plaidml-keras plaidbench
plaidbench keras mobilenet
```
runs as intended.

as well as:
```Python
import os
os.environ["KERAS_BACKEND"] = "plaidml.keras.backend"
import keras
model = keras.Sequential([keras.layers.Dense(units=1, input_shape=[1])])
model.compile(optimizer='sgd',loss='mean_squared_error')
xs = [1, 2, 3, 4, 5]
ys = [2, 4, 6, 8, 10]
model.fit(xs, ys,epochs=10000)
print(model.predict([7]))
```
but:
```Python
import numpy as np
import os
os.environ["KERAS_BACKEND"] = "plaidml.keras.backend"
import keras

# Model / data parameters
num_classes = 10
input_shape = (28, 28, 1)

# the data, split between train and test sets
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

# Scale images to the [0, 1] range
x_train = x_train.astype("float32") / 255
x_test = x_test.astype("float32") / 255
# Make sure images have shape (28, 28, 1)
x_train = np.expand_dims(x_train, -1)
x_test = np.expand_dims(x_test, -1)
print("x_train shape:", x_train.shape)
print(x_train.shape[0], "train samples")
print(x_test.shape[0], "test samples")


# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)
model = keras.Sequential(
    [
        keras.Input(shape=input_shape),
        keras.layers.Conv2D(32, kernel_size=(3, 3), activation="relu"),
        keras.layers.MaxPooling2D(pool_size=(2, 2)),
        keras.layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
        keras.layers.MaxPooling2D(pool_size=(2, 2)),
        keras.layers.Flatten(),
        keras.layers.Dropout(0.5),
        keras.layers.Dense(num_classes, activation="softmax"),
    ]
)

model.summary()
batch_size = 128
epochs = 15

model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])

model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, validation_split=0.1)
score = model.evaluate(x_test, y_test, verbose=0)
print("Test loss:", score[0])
print("Test accuracy:", score[1])
```
throws:

> Traceback (most recent call last):
>   File "privacy/plaidml-test-03.py", line 40, in <module>
>     keras.layers.Dense(num_classes, activation="softmax"),
>   File "privacy\Anaconda\envs\plaidml\lib\site-packages\keras\engine\sequential.py", line 93, in __init__
>     self.add(layer)
>   File "privacy\Anaconda\envs\plaidml\lib\site-packages\keras\engine\sequential.py", line 132, in add
>     'Found: ' + str(layer))
> TypeError: The added layer must be an instance of class Layer. Found: input_1 Placeholder FLOAT32(<tile.Value SymbolicDim UINT64()>, 28, 28, 1)

what I found out so far, is that this error normally occurs when mixing plaidml.keras with tensorflow.keras - which is not the case here. I tried not installing TF, installing TF, running it with tensorflow.keras (which worked, but of course only on cpu instead of GPU) and I am kinda clueless what the problem might be.

**Comment by rescuehero** [comment]
I used: https://keras.io/examples/vision/mnist_convnet/ as example

**Comment by cdaein** [comment]
In your Sequential model, as the first layer, try this:
`keras.layers.InputLayer(input_shape=input_shape)`

or skip the input layer altogether and add input shape to the first Conv2D layer.
`keras.layers.Conv2D(32, kernel_size=(3, 3), activation="relu", input_shape=input_shape)`

**Comment by rescuehero** [comment]
```Python
import numpy as np
import os
os.environ["KERAS_BACKEND"] = "plaidml.keras.backend"
import keras

# Model / data parameters
num_classes = 10
input_shape = (28, 28, 1)

# the data, split between train and test sets
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

# Scale images to the [0, 1] range
x_train = x_train.astype("float32") / 255
x_test = x_test.astype("float32") / 255
# Make sure images have shape (28, 28, 1)
x_train = np.expand_dims(x_train, -1)
x_test = np.expand_dims(x_test, -1)
print("x_train shape:", x_train.shape)
print(x_train.shape[0], "train samples")
print(x_test.shape[0], "test samples")


# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)
model = keras.Sequential(
    [
        keras.layers.Input(shape=input_shape),
        keras.layers.Conv2D(32, kernel_size=(3, 3), activation="relu"),
        keras.layers.MaxPooling2D(pool_size=(2, 2)),
        keras.layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
        keras.layers.MaxPooling2D(pool_size=(2, 2)),
        keras.layers.Flatten(),
        keras.layers.Dropout(0.5),
        keras.layers.Dense(num_classes, activation="softmax"),
    ]
)

model.summary()
batch_size = 128
epochs = 15

model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])

model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, validation_split=0.1)
score = model.evaluate(x_test, y_test, verbose=0)
print("Test loss:", score[0])
print("Test accuracy:", score[1])

```
throws
> Traceback (most recent call last):
>   File "???/plaidml-test-03.py", line 36, in <module>
>     keras.layers.Dense(num_classes, activation="softmax"),
>   File "???\lib\site-packages\keras\engine\sequential.py", line 93, in __init__
>     self.add(layer)
>   File "???\lib\site-packages\keras\engine\sequential.py", line 132, in add
>     'Found: ' + str(layer))
> TypeError: The added layer must be an instance of class Layer. Found: input_1 Placeholder FLOAT32(<tile.Value SymbolicDim UINT64()>, 28, 28, 1)

```Python

import numpy as np
import os
os.environ["KERAS_BACKEND"] = "plaidml.keras.backend"
import keras

# Model / data parameters
num_classes = 10
input_shape = (28, 28, 1)

# the data, split between train and test sets
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

# Scale images to the [0, 1] range
x_train = x_train.astype("float32") / 255
x_test = x_test.astype("float32") / 255
# Make sure images have shape (28, 28, 1)
x_train = np.expand_dims(x_train, -1)
x_test = np.expand_dims(x_test, -1)
print("x_train shape:", x_train.shape)
print(x_train.shape[0], "train samples")
print(x_test.shape[0], "test samples")


# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)
model = keras.Sequential(
    [
        keras.layers.Conv2D(32, kernel_size=(3, 3), activation="relu",input_shape=input_shape),
        keras.layers.MaxPooling2D(pool_size=(2, 2)),
        keras.layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
        keras.layers.MaxPooling2D(pool_size=(2, 2)),
        keras.layers.Flatten(),
        keras.layers.Dropout(0.5),
        keras.layers.Dense(num_classes, activation="softmax"),
    ]
)

model.summary()
batch_size = 128
epochs = 15

model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])

model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, validation_split=0.1)
score = model.evaluate(x_test, y_test, verbose=0)
print("Test loss:", score[0])
print("Test accuracy:", score[1])
```

runs.
but why? :D 
Nonetheless, thank you for the quick fix.
